flink-runtime/src/main/java/org/apache/flink/runtime/io/network/metrics/TimeToConsumeGauge.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.metrics;
+
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.runtime.io.network.partition.consumer.SingleInputGate;
+
+/** Gauge metric measuring the maximal time to consume all buffers of all input gates. */
+public class TimeToConsumeGauge implements Gauge<Long> {
+
+    private final SingleInputGate[] gates;
+
+    public TimeToConsumeGauge(SingleInputGate[] gates) {
+        this.gates = gates;
+    }
+
+    @Override
+    public Long getValue() {
+        long result = Long.MIN_VALUE;
+        for (SingleInputGate gate : gates) {
+            result = Math.max(gate.getLastEstimatedTimeToConsume().toMillis(), result);
```

Comment:
It is because the throughput of each gate is affected by the other gate. 
Basically 
```
throughput_1 = data_1/(busy_1 + busy_2)
```
At the same time gates are consumed in a fair manner. The `getLastEstimatedTimeToConsume` always includes some portion of the time spent on processing the other gate. And in a special case the bigger time will include the entire `lastEstimatedTimeToConsume` of the other gate.

Reply:
I guess theoretically. If were to never see any more records a correct calculation would be to sum all `lastEstimatedTimeToConsume` and divide it by the number of gates. 

Taking the max incorporates an assumption that the "faster" gates will keep on producing at a similar rate, so even after consuming them entirely they will occupy some time with records from a "next portion".

Reply:
thanks for the explanation. Actually, it is obvious that throughput and number of buffers would be divided by each channel.